bolt/lib/Rewrite/RewriteInstance.cpp

```diff
@@ -2439,6 +2439,33 @@ void RewriteInstance::readDynamicRelocations(const SectionRef &Section,
     if (Symbol)
       SymbolIndex[Symbol] = getRelocationSymbol(InputFile, Rel);
 
+    const uint64_t ReferencedAddress = SymbolAddress + Addend;
+    BinaryFunction *Func =
+        BC->getBinaryFunctionContainingAddress(ReferencedAddress);
+
+    if (Relocation::isRelative(RType) && SymbolAddress == 0) {
+      if (Func) {
+        if (!Func->isInConstantIsland(ReferencedAddress)) {
+          if (const uint64_t ReferenceOffset =
+                  ReferencedAddress - Func->getAddress()) {
+            Func->addEntryPointAtOffset(ReferenceOffset);
+          } else if (ReferencedAddress < Func->getAddress()) {
```

Comment:
Since `Func` is discovered using `getBinaryFunctionContainingAddress(ReferencedAddress)`, we should never hit this condition. You can use `ReferencedOffset` unconditionally.

Reply:
I think we can hit the case where the contained address is the same as the function address, leading `ReferenceOffset` to be 0. But the else if condition is redundant you're right. I have removed it.